src/app/screens/LoginScreen/index.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { Image, ImageBackground, KeyboardAvoidingView, Platform, Text } from 'react-native';
+import background from '@assets/bc_inicio.png';
+import logo from '@assets/Group.png';
+import { TextInput, TouchableOpacity } from 'react-native-gesture-handler';
+import actionCreators from '@redux/auth/actions';
+import { isValidEmail, isValidPassword } from '@constants/utils';
+
+import styles from './styles';
+
+function LoginScreen() {
+  const [userValue, onChangeUserText] = useState('');
+  const [passwordValue, onChangePasswordText] = useState('');
+  const [userError, onChangeUserError] = useState(false);
+  const [passwordError, onChangePasswordError] = useState(false);
+  const dispatch = useDispatch();
+  const onPress = () => {
+    onChangeUserError(!isValidEmail(userValue));
+    onChangePasswordError(!isValidPassword(passwordValue));
+    if (isValidEmail(userValue) && isValidPassword(passwordValue)) {
+      dispatch(actionCreators.logIn());
+    }
+  };
```

Comment:
Why are you validating the field only in the `onPress`? You must check in the `onChangeText` and enable the button if there are no errors.
```
const handleChangeUser => (user: string) => {
  onChangeUserError(isValidEmail(user));
  onChangeUserText(user);
}
...
```
Same with the password field and validation
And then, the `onPress`:
```
const onPress = () => dispatch(actionCreators.logIn());`
...
  <TouchableOpacity disabled={!userError && !passwordError} style={styles.button} onPress={onPress}>
    <Text style={styles.buttonText}>Log In!</Text>
  </TouchableOpacity>
```
If you only check error in the `onPress` the error msg doesn't disappear when the user is writing

Reply:
well it's not really defined on the card how it should work, and usually on real projects we are asked to display error texts instead of disabling the submit button, so i made it that way. I'll try disabling the button, seems pretty simple :thumbsup: